### PR TITLE
Issue #344: extract agent process into fat skills

### LINKED ISSUE #344: [Thin Harness A] Extract agent process into fat skills
**Parent**: #343 — Thin Harness, Fat Skills: Architecture Rework

## Customer Problem

Agents carry **55-85% step-by-step process logic** that loads into the context window regardless of whether the current task needs it. This is the core "fat harness" anti-pattern — process knowledge trapped in agent definitions instead of living in on-demand skills.

Additionally, **#345 revealed a salience gap**: cross-cutting pre-response protocols (session-startup, provenance-gate) placed in `copilot-instructions.md` as trigger stubs are easily skipped because agent mode instructions dominate attention. In a live conversation, the session-startup stub at line 57 of `copilot-instructions.md` was not followed — the agent jumped straight into the user's request. These triggers need to move into agent files where they have high salience and plugin distributability.

### What the developer experiences today

- Agent mode instructions load 150-750 lines of process logic on every invocation, most of which may not be relevant to the current task
- Context fills faster; compaction triggers sooner in multi-step sessions
- Cross-cutting startup protocols (session-startup, provenance-gate) are trigger stubs in `copilot-instructions.md` that compete for attention with 300-700 lines of agent mode instructions — and lose

### What good looks like

Agents are thin routing harnesses — identity, delegation, and boundaries. Process logic lives in skills (plugin-distributable, on-demand loaded). Cross-cutting pre-response protocols live in agent mode instructions (high salience). `copilot-instructions.md` is pure orientation (~60 lines).

## User Segments

- **Pipeline users** (full workflow): Pay the highest token tax — every agent hop loads full process logic even when the agent is just routing to a subagent
- **Direct-invocation users**: Load process logic for one agent at a time but still load irrelevant sections
- **Plugin consumers**: Get agents and skills via plugin. Moving triggers from `copilot-instructions.md` to agent files means they get complete agent behavior without maintaining their own trigger stubs

## User Journeys

**Current**: Developer invokes @Code-Smith → 263 lines load (including ~171 lines of process logic the model must process before writing code). Cross-cutting protocols in `copilot-instructions.md` compete with agent mode instructions for attention and sometimes get skipped.

**Target**: Developer invokes @Code-Smith → ~60-100 lines load (routing + frontmatter). Agent loads `implementation-discipline` skill on demand. Pre-response triggers fire reliably from agent mode instructions.

**Edge**: Plugin consumer installs agents and skills. User-invocable agents carry their own pre-response triggers — no `copilot-instructions.md` stubs needed.

### Part 1: Agent Process Extraction (all 14 agents)

For every agent, extract the process/procedure/methodology content into corresponding skills. What stays in agents is routing, identity, and boundaries.

| Agent | Target Skill(s) | Notes |
|-------|----------------|-------|
| Issue-Planner | `plan-authoring` | 6-phase workflow, plan style guide, BDD classification |
| Experience-Owner | `customer-experience` | Upstream framing + downstream CE Gate evidence (two sections in one skill) |
| Research-Agent | `research-methodology` | Research methodology, documentation standards, alternative analysis |
| UI-Iterator | `ui-iteration` | Screenshot workflow, aesthetic evaluation, iteration parameters |
| Test-Writer | Merge into `test-driven-development` | Coverage/mutation standards, quality gates, anti-patterns |
| Refactor-Specialist | `refactoring-methodology` | Proactive hunting, analysis steps, refactoring checklist |
| Code-Smith | `implementation-discipline` | Requirements verification, bad test detection, extraction rules |
| Specification | `specification-authoring` | Template + principles + naming conventions |
| Doc-Keeper | `documentation-finalization` | Update process, quality gates, documentation areas |
| Process-Review | `process-analysis` | Calibration protocol, CE Gate defect analysis, systemic patterns |
| Code-Conductor | `validation-methodology` (partial) | Validation ladder, review reconciliation methodology. **Orchestration/routing stays in agent.** |
| Solution-Designer | `design-exploration` | Stage 2 collaboration pattern, adversarial challenge protocol, end-to-end template |
| Code-Critic | `adversarial-review` | Prosecution perspectives, finding format, section-by-section protocol |
| Code-Review-Response | `review-judgment` | Defense/judge protocol, scoring rubric, verdict categories |

**Skills summary**: 13 new skills created + 1 existing skill updated (`test-driven-development`). Total: 19 → 32 skills.

### Part 2: Cross-Cutting Trigger Migration

Move pre-response trigger stubs from `copilot-instructions.md` into agent files:

| Protocol | Current location | Target location | Agents affected |
|----------|-----------------|-----------------|-----------------|
| Session Startup | copilot-instructions.md stub | `## Process` Step 0 in starting agents | 4 pipeline entry agents (Experience-Owner, Solution-Designer, Issue-Planner, Code-Conductor) |
| Provenance Gate | copilot-instructions.md stub | `## Process` Step 0 in starting agents | Same 4 pipeline entry agents |
| Terminal & Test Hygiene | copilot-instructions.md stub | Code-Conductor (delegates to subagents via skill instructions) | Code-Conductor only |

After migration, `copilot-instructions.md` loses all trigger stubs → becomes pure orientation (~60 lines).

**Accepted gap**: Code-Critic, Code-Review-Response, and UI-Iterator do not get pre-response triggers. These agents are rarely conversation starters; the 4 pipeline entry agents cover 95%+ of conversation starts.

### What Stays in Agents (Thin Agent Template)

```
YAML frontmatter (name, description, tools, handoffs, user-invocable)
  ↓
Persona statement (1-2 lines — "You are a...")
  ↓
## Core Principles (3-5 identity-level principles)
  ↓
## Process
  Step 0 [starting agents only]: Pre-response protocol (session-startup + provenance-gate)
  Load the `{skill-name}` skill and follow its protocol.
  [Conditional skill loading as needed]
  ↓
## Boundaries (DO/DON'T lists — routing-level constraints)
  [Safety-critical operational constraints stay here, not in the skill]
```

### Part 3: Architecture Rule Clarification

Amend `architecture-rules.md` forbidden rule:

**Current**: "Skills must NOT contain agent logic — they provide knowledge, not orchestration"

**Amended**: "Skills must NOT contain agent orchestration — routing decisions, handoff logic, subagent delegation, tool selection, commit policy. Skills carry domain knowledge, methodology, and operational protocols."

**Concrete examples** (included in amendment):
- **Orchestration** (stays in agents): Subagent routing, handoff decisions, tool selection, commit policy, Code-Conductor's step execution loop, skill mapping table
- **Methodology** (moves to skills): TDD workflow, research documentation standards, prosecution perspectives, validation ladder protocol, calibration scoring

### Additional Scope from Adversarial Challenge

- **Code-Conductor's skill mapping table** must be updated with new skill references for all subagent delegations
- **Pester contract test** (`session-startup-wording-contract.Tests.ps1`) must be updated — it currently requires `copilot-instructions.md` to retain the session-startup trigger stub
- **Safety-critical boundaries** (e.g., Research-Agent's CAN DO / CANNOT DO lists) stay in agent `## Boundaries` sections, not in skills
- **Conditional skill loading** (e.g., Test-Writer: "When UI work, also load `ui-testing`") stays in agent `## Process` sections — it's routing logic
- **Questioning policy** rules (the mandate to use `askQuestions`) stay in agents; question-framing methodology moves to skills
- **This is architectural restructuring** under parent #343 "Thin Harness" — not incremental extraction per guidance-complexity D6 criteria

## Design Decisions

- **D1 — Skill naming**: Use descriptive methodology names (`refactoring-methodology`), not agent names (`refactor-specialist-process`), so skills are reusable across agents
- **D2 — Merge vs. create**: Where a corresponding skill already exists (e.g., `test-driven-development` for Test-Writer), merge the agent's process into the existing skill rather than creating a duplicate. Experience-Owner's upstream framing and downstream CE Gate evidence merge into a single `customer-experience` skill with two sections.
- **D3 — Agent personality preservation**: Each agent's voice and stance ("proactive hunter", "customer advocate") stays in the agent file as a 1-2 line persona statement, not in the skill
- **D4 — Trigger-stub salience** (from #345 lesson): Cross-cutting pre-response protocols live in agent `## Process` Step 0 (high salience), not `copilot-instructions.md` stubs (low salience). Agent files are plugin-distributable, solving the downstream propagation problem. Scoped to 4 pipeline entry agents; 3 non-starting user-invocable agents accept the gap.
- **D5 — Separation of concerns over line counts**: Extraction targets routing-vs-process separation, not an arbitrary line count. Some agents may land at 40 lines, others at 120+, depending on routing complexity. The goal is that process logic lives in skills, not that agents hit a specific number.
- **D6 — Architecture rules clarification**: Amend "skills must not contain agent logic" to distinguish orchestration (stays in agents) from methodology/protocols (moves to skills). Include concrete examples to reduce ambiguity for downstream consumers.
- **D7 — Standardized delegation block**: Each agent gets a `## Process` section with "Load the `{skill-name}` skill (`.github/skills/{skill-name}/SKILL.md`) and follow its protocol." Consistent pattern across all agents.
- **D8 — Pre-response trigger placement**: Process Step 0 in the 4 starting agents (Experience-Owner, Solution-Designer, Issue-Planner, Code-Conductor). Terminal-hygiene trigger moves to Code-Conductor only (it delegates to subagents).
- **D9 — Code-Conductor partial extraction**: Only methodology portions (validation ladder, review reconciliation) extract. Orchestration (subagent routing, step execution, commit policy, skill mapping) stays. The agent remains the largest because orchestration is its core purpose.
- **D10 — Safety constraints stay in agents**: Operational constraints that prevent wrong behavior (Research-Agent CAN/CANNOT DO, Code-Conductor commit rules) stay in the agent `## Boundaries` section, not in skills. These are always-loaded safety guards.

### Rejected Alternatives

- **Keep triggers in copilot-instructions.md (status quo)**: Rejected — #345 demonstrated the salience gap in a live conversation. Plugin consumers also can't rely on `copilot-instructions.md` stubs.
- **Meta-skill resolver**: Single skill dispatching to session-startup and provenance-gate. Rejected — adds indirection (skills loading skills) without clear benefit over direct agent triggers.
- **Merge ui-iteration into frontend-design**: Rejected — different concerns. `frontend-design` = design principles; UI-Iterator = screenshot-driven iteration process. Merging would bloat the skill and mix "what makes good design" with "how to iterate."
- **Structured YAML frontmatter skills field**: Add `skills:` to agent YAML. Rejected — no existing precedent, may affect VS Code parsing, doesn't capture *when* to load each skill.
- **Agent-voice delegation**: Let each agent express delegation in its own personality. Rejected — harder to validate structurally; inconsistent patterns across agents.

### S1 — Specialist agent loads skill on demand (Functional)

Invoke Code-Smith with an implementation task. The agent loads `implementation-discipline` skill and follows its process. The agent file itself contains no implementation steps.

### S2 — Existing skills absorb agent process without duplication (Functional)

Test-Writer's process merges into the existing `test-driven-development` skill. No duplicate RED-GREEN-REFACTOR guidance exists across the skill and agent.

### S3 — Session startup fires reliably from agent trigger (Functional)

Start a new conversation with @Experience-Owner. Step 0 in the agent's Process section triggers session-startup skill loading. The session cleanup detector runs before any response — unlike the pre-#344 `copilot-instructions.md` stub, which was skipped in a live conversation.

### S4 — Plugin consumer gets complete agent behavior (Functional)

A plugin consumer installs agents and skills. User-invocable agents carry their own pre-response triggers. No `copilot-instructions.md` stubs needed for startup or provenance protocols.

### S5 — Code-Conductor retains orchestration after partial extraction (Functional)

Code-Conductor's step execution loop, subagent routing, and commit policy remain in the agent. Only the validation ladder and review reconciliation methodology extract to `validation-methodology` skill.

## Customer Surface

Terminal-based: invoke agents via Copilot chat, verify behavior matches current baseline. `quick-validate.ps1` + Pester as structural gates. Manual conversation testing for S3 (session-startup fires from agent trigger).

## Design Intent

Agents should read as identity cards — "who am I, what do I delegate to, and what are my boundaries." Process knowledge loads on demand from skills. Cross-cutting protocols fire reliably because they live where the model is paying attention: agent mode instructions.

## CE Gate Readiness

1. **Surface**: Terminal-based (Pester, quick-validate, manual conversation testing)
2. **Tool availability**: All tools available locally
3. **Manual fallback**: Start new conversations with user-invocable agents and verify session-startup, provenance-gate, and skill loading work correctly after extraction
4. **Scenarios**: S1 (functional), S2 (functional), S3 (functional), S4 (functional), S5 (functional)

## Adversarial Design Challenge Summary

3-pass prosecution completed (2 design-review passes, 1 product-alignment pass). ~25 raw findings deduplicated to 17 unique items. **8 incorporated**, 3 acknowledged by user decision, 6 dismissed.

### Incorporated findings

| # | Finding | Severity | Action |
|---|---------|----------|--------|
| 1 | Pester contract test (`session-startup-wording-contract.Tests.ps1`) requires copilot-instructions.md to retain the session-startup stub | Critical | Update test to target agent files or new skill-loading pattern |
| 2 | Process-Review (~630+ lines) excluded without rationale — 2nd largest agent | Medium | Added to extraction table (now all 14 agents in scope) |
| 3 | Architecture-rules D6 amendment needs concrete examples of orchestration vs methodology | Medium | Include examples in the amendment |
| 4 | Code-Conductor's skill mapping table + Process-Review sync table need updating | Medium | Added to scope |
| 5 | Safety-critical operational constraints must stay in agents, not move to skills | Medium | Addressed in D10 |
| 6 | Conditional skill loading (Test-Writer UI/BDD) stays in agent Process section | Medium | Noted as routing logic |
| 7 | Relationship to guidance-complexity D6 extraction criteria needs acknowledgment | Medium | Noted as architectural restructuring under #343 |
| 8 | D6 amendment affects downstream consumers — needs concrete examples | Low | Combined with finding #3 |

### User-confirmed decisions

| # | Finding | Decision |
|---|---------|----------|
| 9 | Trigger scoping to 4 agents is behavioral regression for 3 non-starting agents | Accepted — 4 pipeline entry agents cover 95%+ of starts |
| 10 | Line counts are directional, not precise — actual counts differ from estimates | Accepted — counts are directional; implementation discovers actuals |
| 11 | All 14 agents in scope including Code-Conductor (partial extraction) | Confirmed |

### Dismissed findings

| # | Finding | Rationale |
|---|---------|-----------|
| 12 | customer-experience skill merges two lifecycle phases | User approved merge; skill uses two sections |
| 13 | Terminal-hygiene scoping to Code-Conductor leaves other agents uncovered | Code-Conductor delegates to subagents via skill instructions |
| 14 | Guidance-complexity ceiling may need recalibration | Thin agents being under ceiling is the desired state |
| 15 | D5 vs guidance-complexity relationship unclear | Compatible — directives ≠ lines |
| 16 | No #345 retrospective referenced | #345 salience lesson IS referenced as D4 motivation |
| 17 | New skills must pass quick-validate frontmatter checks | Already in acceptance criteria |

## Testing Scope

| Change Type | Unit | Integration | E2E |
|-------------|------|-------------|-----|
| Skill creation (13 new) | quick-validate frontmatter | No | No |
| Skill merge (1 existing) | quick-validate + existing Pester | No | No |
| Agent slimming (14 agents) | quick-validate + existing Pester | No | Manual conversation test |
| Trigger migration (4 agents) | Pester contract test update | No | Manual conversation test |
| Architecture-rules amendment | Manual review | No | No |

## Acceptance Criteria

- [ ] All 14 agents slimmed to routing shells (process extracted to skills, orchestration/safety stays)
- [ ] 13 new skills created: `plan-authoring`, `customer-experience`, `research-methodology`, `ui-iteration`, `refactoring-methodology`, `implementation-discipline`, `specification-authoring`, `documentation-finalization`, `process-analysis`, `validation-methodology`, `design-exploration`, `adversarial-review`, `review-judgment`
- [ ] 1 existing skill updated: `test-driven-development` (Test-Writer process merged)
- [ ] New skills include required frontmatter (`Use when`, `DO NOT USE FOR:`, `## Gotchas`)
- [ ] Pre-response triggers (session-startup, provenance-gate) moved from `copilot-instructions.md` to 4 starting agents (Process Step 0)
- [ ] Terminal-hygiene trigger moved from `copilot-instructions.md` to Code-Conductor
- [ ] `copilot-instructions.md` trigger stubs removed (file becomes pure orientation, ~60 lines)
- [ ] `architecture-rules.md` amended with orchestration vs. methodology distinction + concrete examples
- [ ] `plugin.json` updated with 13 new skill entries (32 total)
- [ ] Code-Conductor skill mapping table updated with new skill references
- [ ] Pester contract test (`session-startup-wording-contract.Tests.ps1`) updated for new trigger locations
- [ ] Safety-critical operational constraints remain in agent `## Boundaries` sections
- [ ] Artifact counts updated across all files (skills: 32, instructions: 6)
- [ ] `quick-validate.ps1` passes (agent count = 14, no broken references)
- [ ] Pester tests pass
- [ ] No agent interface changes (names, tools, handoffs unchanged)

### PR DESCRIPTION
**Pull request by Grimblaz**
## Summary

Implements issue #344's thin-agents / fat-skills restructuring.

This PR moves reusable methodology out of agent files and into on-demand skills, keeps orchestration and safety boundaries in the owning agents, migrates the portable session-startup / provenance-gate triggers into the four pipeline-entry agents, and updates the shipped inventory and downstream-facing docs to the new 14-agent / 32-skill shape.

## Changed Files

- Agent boundaries and trigger ownership: `.github/agents/*.agent.md` including `Code-Conductor`, `Experience-Owner`, `Issue-Planner`, `Code-Smith`, `Test-Writer`, and the other slimmed agents.
- New reusable methodology skills: `adversarial-review`, `customer-experience`, `design-exploration`, `documentation-finalization`, `implementation-discipline`, `plan-authoring`, `process-analysis`, `refactoring-methodology`, `research-methodology`, `review-judgment`, `specification-authoring`, `ui-iteration`, and `validation-methodology` under `.github/skills/`.
- Existing skill merge/update: `.github/skills/test-driven-development/SKILL.md`.
- Contract and architecture surfaces: `.github/architecture-rules.md`, `.github/copilot-instructions.md`, `.github/scripts/Tests/session-startup-wording-contract.Tests.ps1`, and `.github/scripts/Tests/handoff-persistence-contract.Tests.ps1`.
- Inventory and consumer surfaces: `.github/plugin/plugin.json`, `.github/plugin/marketplace.json`, `README.md`, `CUSTOMIZATION.md`, `.github/skills/README.md`, `Documents/Design/skills-framework.md`, and the example `copilot-instructions.md` files.

## Validation Evidence

- `pwsh -NoProfile -NonInteractive -Command "Invoke-Pester .github/scripts/Tests/ -Output Minimal"` -> `Tests Passed: 386, Failed: 0, Skipped: 1`.
- `pwsh -NoProfile -NonInteractive -File .github/scripts/quick-validate.ps1` -> `Quick-validate: 9/9 checks passed`.
- `pwsh -NoProfile -NonInteractive -Command "Invoke-Pester .github/scripts/Tests/bdd-scenario-contract.Tests.ps1,.github/scripts/Tests/plan-approval-prompt-contract.Tests.ps1 -Output Minimal"` -> `Tests Passed: 39, Failed: 0, Skipped: 0`.
- Targeted markdownlint passes completed on the touched skill/doc slices used during extraction and review-fix work.

## CE Gate

✅ CE Gate passed — intent match: strong

Scenarios exercised:
- `S1` Passed: Code-Smith delegates reusable implementation workflow to `implementation-discipline` and no longer carries embedded implementation procedure.
- `S2` Passed: `test-driven-development` holds the merged Test-Writer methodology without duplicate routing logic in `Test-Writer`.
- `S3` Passed: the four pipeline-entry agents carry `session-startup` and `provenance-gate` trigger text, and both contract suites passed.
- `S4` Passed: every Step 8 terminal-capable agent carries `terminal-hygiene` guidance.
- `S5` Passed: Code-Conductor retains orchestration ownership after partial extraction.
- `S6` Passed: `plugin.json` resolves to 32 skill directories on disk and count references are updated across the named docs and examples.

Process-Review: no systemic CE gap found. A session-level workflow note was recorded instead: high-output validation runs can stall the shared VS Code terminal, so targeted checks, `-Output Minimal`, and fresh/background terminals were used successfully as the mitigation.

## Adversarial Review Scores

| Stage | Prosecutor | Defense | Judge rulings |
| --- | --- | --- | --- |
| Code Review | 17 pts (4 sustained) | 2 pts (2 disproved, 2 rejected) | 6 rulings |
| CE Review | 0 pts (0 sustained) | 0 pts (0 disproved, 0 rejected) | 0 rulings |
| Post-fix Review | 0 pts (0 sustained) | 0 pts (0 disproved, 0 rejected) | 0 rulings |

## Prosecution Depth Summary

Prosecution depth: all full (fallback — aggregate script unavailable)

## Process Gaps Found

- Validation noise, not product behavior, was the main execution risk in this session. Repeated high-output terminal runs made VS Code stall-prone.
- Guardrail used in this run: keep inner-loop validation targeted, use `Invoke-Pester ... -Output Minimal`, inspect captured output instead of immediately rerunning noisy commands, and move broad validation to a fresh/background terminal after one or two high-output passes.
- No separate process issue was opened from this run.

## Pipeline Metrics

Closes #344

### CODE REVIEW

File: .github/skills/research-methodology/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+4. Compare viable approaches against project constraints, conventions, and maintenance cost.
+5. Reduce alternatives to one recommended approach and remove discarded paths from the final notes.
+
+## Evidence Collection
+
+### Internal Research
+
+- Read the closest owning files before broad exploration
+- Search for repeated patterns, not isolated snippets
+- Check usage sites to understand how a pattern is actually applied
```
[the remark this answers is not in the document]
Comment (Grimblaz) [reply .github/skills/research-methodology/SKILL.md:47]: Won't fix — `#codebase`, `#search`, `#usages` are VS Code Chat participant syntax, which is platform-specific and fragile. Skills are designed to be tool-agnostic methodology; the owning agent controls tool selection. Adding platform-specific tool names here would violate the thin-agent/fat-skill boundary.

File: .github/skills/validation-methodology/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+### Pre-Review Gate
+
+Before calling the prosecutor, run the repository validation commands that clear trivial lint, typecheck, or harness failures. The review cycle should spend its budget on substantive defects, not on already-known validation noise.
+
+### Prosecution Depth Setup
+
+Before composing pass prompts:
+
+1. Run `aggregate-review-scores.ps1` and capture `prosecution_depth:` output.
```
[the remark this answers is not in the document]
Comment (Grimblaz) [reply .github/skills/validation-methodology/SKILL.md]: ✅ Fixed in 63e911b — updated to full path `.github/scripts/aggregate-review-scores.ps1`.

File: .github/agents/Code-Conductor.agent.md [hunk 3]
```diff
@@ -105,10 +105,25 @@ Quick checklist before declaring mode for a step:
...
+For terminal and validation execution guardrails, load `.github/skills/terminal-hygiene/SKILL.md`.
+
 ## Core Workflow
 
-<!-- markdownlint-disable-next-line MD029 -->
+Any future pre-response trigger step runs **before** the Core Workflow, stays outside the numbered workflow list, and does not renumber, replace, or subsume Step 0. Issue Transition remains Step 0 and the first numbered workflow step after any pre-response trigger handling completes.
 
+<!-- markdownlint-disable-next-line MD029 -->
 0. **Issue Transition (Step 0, before implementation)**:
    - Cleanup note: The `.github/copilot-instructions.md` "Session Startup Check" detects stale tracking files from merged branches and prompts you at the start of your next conversation — cleanup requires one confirmation. If stale artifacts persist, run `$copilotRoot = if ($env:COPILOT_ORCHESTRA_ROOT) { $env:COPILOT_ORCHESTRA_ROOT } else { $env:WORKFLOW_TEMPLATE_ROOT }; pwsh "$copilotRoot/.github/scripts/post-merge-cleanup.ps1" -IssueNumber {N} -FeatureBranch feature/issue-{N}-description` directly (only if `$copilotRoot` is non-empty — requires `COPILOT_ORCHESTRA_ROOT` or `WORKFLOW_TEMPLATE_ROOT` to be set).
```
[the remark this answers is not in the document]
Comment (Grimblaz) [reply .github/agents/Code-Conductor.agent.md]: ✅ Fixed in 63e911b — updated to reference the `session-startup` skill (loaded by pipeline-entry agents) instead of the removed `copilot-instructions.md` section.

File: .github/skills/adversarial-review/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+- Native command exit-code checks at boundaries
+- Cross-references to authoritative enumerated values
+- PowerShell and pipeline semantics that preserve intended types
+
+For markdown-only command guidance, audit:
+
+- Runnable commands from repo root
+- Self-match hazards in grep-based validations
+- Correct post-change counts and expectations
+- Preference for built-in VS Code tools over terminal-first read-only guidance when an equivalent exists
```
[the remark this answers is not in the document]
Comment (Grimblaz) [reply .github/skills/adversarial-review/SKILL.md:133]: Won't fix — line 133 already contains the summary: "Preference for built-in VS Code tools over terminal-first read-only guidance when an equivalent exists". The specific exception list is an agent-level detail that belongs in Code-Conductor's terminal-hygiene skill, not in the reusable adversarial-review methodology.

File: .github/skills/research-methodology/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+- Verify repository conventions from architecture and instruction surfaces before recommending structural changes
+
+### External Research
+
+- Prefer official documentation, standards, or authoritative repositories
+- Record why the source is relevant, not just what it says
```
[the remark this answers is not in the document]
Comment (Grimblaz) [reply .github/skills/research-methodology/SKILL.md:53]: Won't fix — same rationale as above. `#fetch` and `#githubRepo` are platform-specific tool references that don't belong in reusable methodology. The skill describes what to do; the agent decides which tools to use.

File: .github/skills/adversarial-review/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+### 1. Architecture
+
+Apply when runtime code, scripts, or runtime configuration changed.
+
+Check:
+
+- Architecture-rule compliance and layer direction
+- Integration wiring for new components
+- Data integration for newly introduced fields, constants, and maps
+- Domain-alignment mismatches across validators, parsers, and converters
```
[the remark this answers is not in the document]
Comment (Grimblaz) [reply .github/skills/adversarial-review/SKILL.md]: ✅ Fixed in 63e911b — expanded the domain-alignment line to include actionable peer-identification methodology: "identify peers via field-name grep, plan consultation for aliases, and call-chain tracing".

File: .github/skills/implementation-discipline/SKILL.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+If a requirement is missing from tests but clearly part of the requested behavior, implement it anyway and call out the missing coverage.
+
+## Delegation Instead Of Duplication
+
+When a new file or class needs logic that already exists:
+
+1. Search for the existing formula, mapping, or validation first.
+2. Inject the dependency and call it instead of copying behavior.
+3. Use composition, strategies, or pipelines rather than parallel duplicate implementations.
```
[the remark this answers is not in the document]
Comment (Grimblaz) [reply .github/skills/implementation-discipline/SKILL.md:57]: Won't fix — `CalculationService` is a domain-specific example. The skill already states the principle: "Search for the existing formula, mapping, or validation first. Inject the dependency and call it instead of copying behavior. Use composition, strategies, or pipelines rather than parallel duplicate implementations." This is sufficient without a consumer-domain code example.

File: .github/skills/test-driven-development/SKILL.md [hunk 8]
```diff
@@ -46,6 +49,48 @@ Quality gates are enforced in priority order:
 - **Test rules, not formulas**: "higher values produce larger results" not exact arithmetic
 - **Keep test files focused**: Split by behavior if tests become unwieldy
 
+## Behavior Quality Standards
+
+- Use business-language test names that describe expected outcomes, not method names or internal branches
+- Prefer one observable behavior per test with a clear Arrange-Act-Assert structure
+- Cover realistic edge cases and requirement boundaries, not speculative permutations
+- Use parameterized tests for data tables, formulas, or repeated rule checks that differ only by inputs and outcomes
+- Treat every added test as a requirement statement; avoid "vibe coding" assertions that do not map to a concrete behavior
```
[the remark this answers is not in the document]
Comment (Grimblaz) [reply .github/skills/test-driven-development/SKILL.md:58]: Won't fix — `highVit.health > lowVit.health` is a consumer-domain (game) example that doesn't belong in a reusable template skill. The principle is already stated clearly: "Treat every added test as a requirement statement; avoid 'vibe coding' assertions that do not map to a concrete behavior." Consumer repos supply domain-specific examples via their own `copilot-instructions.md`.
